internal/deprecations.go

```diff
@@ -278,8 +282,16 @@ func NormalizeClusterSpec(cluster *fdbtypes.FoundationDBCluster, options Depreca
 			cluster.Spec.AutomationOptions.Replacements.FailureDetectionTimeSeconds = pointer.Int(1800)
 		}
 
-		cluster.Spec.MainContainer.ImageConfigs = append(cluster.Spec.MainContainer.ImageConfigs, fdbtypes.ImageConfig{BaseImage: "foundationdb/foundationdb"})
-		cluster.Spec.SidecarContainer.ImageConfigs = append(cluster.Spec.SidecarContainer.ImageConfigs, fdbtypes.ImageConfig{BaseImage: "foundationdb/foundationdb-kubernetes-sidecar", TagSuffix: "-1"})
+		if cluster.Spec.UseUnifiedImage == nil {
+			cluster.Spec.UseUnifiedImage = pointer.Bool(useUnifiedImage)
+		}
```

Comment:
```suggestion
		cluster.Spec.UseUnifiedImage = useUnifiedImage
```

`useUnifiedImage` will be already `false` if `cluster.Spec.UseUnifiedImage` is not set otherwise we just overwrite `cluster.Spec.UseUnifiedImage` with the same value again, so that simplifies things.